src/pymedphys/_labs/paulking/profile.py

```diff
@@ -81,42 +110,40 @@ def interactive(self):
 
 
 class DoseProfile(Dose1D):
-    def __init__(self, *args, metadata={}, x=[], data=[], **kwargs):
-        try:  # DoseProfile( x=[0,2,4], data=[1,3,5] )
-            x = kwargs.pop('x')
-            data = kwargs.pop('data')
-        except (KeyError):
-            pass
-
-        try:  # DoseProfile( [0,2,4], [1,3,5] )
-            assert (len(args) == 2)
-            assert len(args[0]) == len(args[1])
-            x = args[0]
-            data = args[1]
-        except (AssertionError, IndexError):
-            pass
-
-        try:  # DoseProfile( [(0,1),(2,3),(4,5) ]
-            assert len(args) == 1
-            assert len(args[0][0]) == 2
-            x = list(list(zip(*args[0]))[0])
-            data = list(list(zip(*args[0]))[1])
-        except (AssertionError, IndexError):
-            pass
-
-        super().__init__(x, data)
-
+    def __init__(self, x, data, metadata={}):
         self.metadata = metadata
+        super().__init__(x, data)
 
     def __len__(self):
         assert len(self.x) == len(self.data)
         return len(self.x)
 
+    def from_tuples(self, list_of_tuples, metadata={}):
```

Comment:
Instead of making this happen inplace this might be better as a "class method". That way it can be used to create the class from the beginning.

Reply:
It is certainly my preference that if possible we avoid inplace editing there are often cleaner ways to do it that are easier to reason about when trying to understand code. It's not a problem here yet, but once a range of things all happen in place the complexity of working out what's going on goes up multiplicatively.

Reply:
Fair enough, though I expect it will end up being useful.